### PR TITLE
JAMES-2544 ManagementCredentials info should be apart of RabbitMQConfiguration

### PR DESCRIPTION
**Pull request by trantienduchn**
just a small proposal. `RabbitMQManagementCredentials` contains `user` and `password` which can be loaded from configuration file

### CODE REVIEW

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java [hunk 1]
```diff
@@ -28,6 +32,57 @@
 import com.google.common.base.Strings;
 
 public class RabbitMQConfiguration {
+
+    public static class ManagementCredentials {
+
+        static final char[] DEFAULT_PASSWORD = "guest".toCharArray();
+        static final String DEFAULT_USER = "guest";
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: From serveral web frameworks I worked with before, they provide default mysql credentials if I don't specify it inside configuration file. It makes sense to me.

[the remark this answers is not in the document]
Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: IMO James should be preventing a user from using a default login/password without knowing it. We should rather fail for safety reasons.

The downside is that we may require a bit more configuration to get it working.

[the remark this answers is not in the document]
Comment (mbaechler) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: > From serveral web frameworks I worked with before, they provide default mysql credentials if I don't specify it inside configuration file. It makes sense to me.

Maybe that's acceptable for a framework but for a mail server, I don't think so.

[the remark this answers is not in the document]
Comment (trantienduchn) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: 👍

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java [hunk 2]
```diff
@@ -35,16 +35,20 @@
 
     public static class ManagementCredentials {
 
-        static final char[] DEFAULT_PASSWORD = "guest".toCharArray();
         static final String DEFAULT_USER = "guest";
+        static final String DEFAULT_PASSWORD_STRING = "guest";
```
Comment (chibenwa) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: Are the DEFAULT_PASSWORD_STRING & DEFAULT_USER still required?

Comment (trantienduchn) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: I fixed. They are required. those constants are used in tests

Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConfiguration.java]: Can we then locate them where appropriate in tests?

File: backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/RabbitMQConfigurationTest.java [hunk 3]
```diff
@@ -125,19 +129,55 @@ void fromShouldReturnTheConfigurationWhenRequiredParametersAreGiven() {
...
         assertThat(RabbitMQConfiguration.from(configuration))
             .isEqualTo(RabbitMQConfiguration.builder()
                 .amqpUri(URI.create(amqpUri))
                 .managementUri(URI.create(managementUri))
+                .managementCredentials(DEFAULT_MANAGEMENT_CREDENTIAL)
                 .build());
     }
 
+    @Test
+    void fromShouldThrowWhenNotGiven() {
```
Comment (mbaechler) [inline backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/RabbitMQConfigurationTest.java]: when management credentials not given ?

File: backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/RabbitMQConfigurationTest.java [hunk 4]
```diff
@@ -125,19 +129,55 @@ void fromShouldReturnTheConfigurationWhenRequiredParametersAreGiven() {
...
+        String managementUri = "http://james:james@rabbitmq_host:15672/api/";
+        configuration.addProperty("management.uri", managementUri);
+
+        assertThatThrownBy(() -> RabbitMQConfiguration.from(configuration))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessage("You need to specify the management.user property as username of rabbitmq management admin account");
+    }
+
+    @Test
+    void fromShouldReturnCustomValueWhenGiven() {
```
Comment (mbaechler) [inline backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/RabbitMQConfigurationTest.java]: same thing, we should mention "management credentials"

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**